dhall/src/Dhall/Eval.hs

```diff
@@ -527,6 +528,15 @@ eval !env t =
                             | Just f  <- Dhall.Map.lookup k m -> maybe f (vApp f) mt
                             | otherwise -> error errorMsg
                           (x, y, ma) -> VMerge x y ma
+    ToMap x ma       -> case (evalE x, evalE <$> ma) of
+                          (VRecordLit m, Just (VList t)) | null m ->
+                              VListLit (Just t) (Dhall.Map.foldMapWithKey entry m)
+                          (VRecordLit m, _) ->
+                              VListLit Nothing (Dhall.Map.foldMapWithKey entry m)
```

Comment:
`mconcat`ting a singleton for each element seems fairly inefficient to me. I suspect it would be better to do something like

```
Data.Sequence.fromList . ... . Dhall.Map.toList
```